docs/pmm-upgrade/migrating_from_pmm_2.md

```diff
@@ -13,45 +13,100 @@ Before upgrading to PMM 3, ensure your PMM 2 Server is running the latest versio
 
 ## Step 2: Migrate PMM 2 Server to PMM 3
 
-Follow these manual steps to upgrade your PMM 2 Server to PMM 3:
-{.power-number}
+=== "For Docker Volume"
+
+    Follow these manual steps to upgrade your PMM 2 Server to PMM 3:
+    { .power-number}
 
-1. Pull the new PMM 3 Server Docker image:
+    1. Stop all PMM Server services:
 
     ```sh
-    docker pull percona/pmm-server:3
+    docker exec -t <pmm-server> supervisorctl stop all
     ```
 
-2. Stop all services of the PMM 2 Server container:
+    2. Transfer `/srv` directory ownership:
 
     ```sh
-    docker exec -t pmm-server supervisorctl stop all
+    docker exec -t <pmm-server> chown -R pmm:pmm /srv
     ```
 
-3. Transfer ownership of all directories and files in the `/srv` directory to the pmm user:
-
+    3. List and note down your Docker volume:
+   
     ```sh
-    docker exec -t pmm-server chown -R pmm:pmm /srv
+    [ec2-user@ip-10-178-1-6 ~]$ docker inspect -f '{{range .Mounts}}{{if eq .Type "volume"}}{{.Name}}{{"\n"}}{{end}}{{end}}' pmm-server
+    pmm-server-data
     ```
 
-4. Stop and remove the PMM 2 Server container:
+    4. Stop and remove existing container:
 
     ```sh
     docker stop pmm-server && docker rm pmm-server
     ```
 
-5. Run a new container based on the PMM 3 image, ensuring you pass the same pmm-data volume:
+    5. Pull PMM 3 Server image:
+
+    ```sh
+    docker pull percona/pmm-server:3
+    ```
+
+    6. Run new container with existing volume:
+   
+    ```sh
+    docker run -d -v pmm-server-data:/srv -p 80:8080 -p 443:8443 --name pmm-server --restart always percona/pmm-server:3
+    ```
+
+=== "For Data Container"
```

Comment:
Same here. I'd rename it to something like "For those running PMM v2 with docker data containers" to clarify that these steps should be executed if PMM2 server is running using docker data-container